camel-core/src/main/java/org/apache/camel/component/log/LogEndpoint.java

```diff
@@ -141,7 +142,13 @@ protected Processor createLogger() throws Exception {
             Long groupDelay = getGroupDelay();
             answer = new ThroughputLogger(camelLogger, this.getCamelContext(), getGroupInterval(), groupDelay, groupActiveOnly);
         } else {
-            answer = new CamelLogProcessor(camelLogger, localFormatter, getMaskingFormatter());
+            CamelLogProcessor processor = new CamelLogProcessor(camelLogger, localFormatter, getMaskingFormatter());
```

Comment:
Maybe add a ctr parameter that takes the Set<LogListener> so you can call this instead of adding them afterwards with the forEeach